### PR TITLE
FISH-6234 Servlet API upgrade to v6.0.0

### PR DESCRIPTION
**Pull request by pdudits**
## Description

This serves as workaround to allow implementation of Servlet 6 to be done while other upstream dependencies still depend on Servlet 5. It takes server-api 6.0 and reexports it as *both* Servlet 6.0.0 and 5.1.9, which is made-up synthetic version.

This allows server to start, but potentially break at runtime.

This is transitional solution that needs to be removed before final release (and marked with terms "transitional" in the pom files so that we can locate these changes later).

### Testing Performed
Admin console starts up with repackaged servlet 6 provided in modules directory.

### CODE REVIEW

**Review by jGauravGupta — APPROVED** [review]

**Comment by jGauravGupta** [comment]
[jakarta-servlet-tck-6.0.0](https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-servlet-tck-6.0.0.zip) Number of tests completed:  1694 (1627 passed, 67 failed, 0 with errors)

**Comment by jGauravGupta** [comment]
Current status:
Number of tests completed:  1694 (1662 passed, 31 failed, 1 with errors)

**Comment by jGauravGupta** [comment]
Current status:
Number of tests completed:  1694 (1685 passed, 9 failed, 0 with errors)

File: appserver/web/web-core/src/main/java/org/apache/catalina/connector/ServletConnectionImpl.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Override
+    public String getProtocol() {
+        return protocol == null ? "unknown" : this.protocol.getProtocolString();
+    }
+
+    @Override
+    public String getProtocolConnectionId() {
+        throw new UnsupportedOperationException("Not supported yet.");
```
Comment (pdudits) [inline appserver/web/web-core/src/main/java/org/apache/catalina/connector/ServletConnectionImpl.java:71]: woudln't `connectionId` be good enough for start?

File: appserver/packager/appserver-core/pom.xml [hunk 2]
```diff
@@ -263,7 +263,7 @@
 <!--        </dependency>-->
         <dependency>
             <groupId>fish.payara.server.internal.packager</groupId>
-            <artifactId>jakarta.servlet-api-transition</artifactId>
+            <artifactId>jakarta.servlet-api</artifactId>
             <version>${project.version}</version>
```
Comment (pdudits) [inline appserver/packager/appserver-core/pom.xml:267]: why do we move back to original api, but other references i. e. admingui were not updated?

File: appserver/packager/external/jakarta.servlet.jsp-api-transition/pom.xml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>fish.payara.server.internal.packager</groupId>
+        <artifactId>external</artifactId>
+        <version>6.2022.1.Alpha3-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>jakarta.servlet.jsp-api</artifactId>
```
Comment (pdudits) [inline appserver/packager/external/jakarta.servlet.jsp-api-transition/pom.xml:54]: I'd expect `-transition` suffix so we can recognize these in modules directory

File: nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/Http.java [hunk 4]
```diff
@@ -112,7 +112,7 @@ public interface Http extends ConfigBeanProxy, PropertyBag {
     float HTTP2_CLEAN_PERCENTAGE = 0.5f;
     int HTTP2_CLEAN_FREQUENCY_CHECK = 50;
     boolean HTTP2_DISABLE_CIPHER_CHECK = false;
-    boolean HTTP2_PUSH_ENABLED = false;
+    boolean HTTP2_PUSH_ENABLED = true;
```
Comment (pdudits) [inline nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/Http.java:115]: If I remember right we ship with http2 push disabled on purpose; server should be configured with push enabled for tck

Comment (jGauravGupta) [reply nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/dom/Http.java]: Added configuration setting to TCK Runer:
````
<exec executable="${glassfish.asadmin}" dir="${glassfish.home}/glassfish/bin">
  <arg value="set" />
  <arg value="configs.config.server-config.network-config.protocols.protocol.http-listener-1.http.http2-push-enabled=true" />
</exec>
````

File: appserver/web/web-core/src/main/java/org/apache/catalina/core/SessionCookieConfigImpl.java [hunk 5]
```diff
@@ -38,33 +38,39 @@
...
 import java.util.Map;
+import java.util.TreeMap;
 
 /**
- * Class that may be used to configure various properties of cookies 
- * used for session tracking purposes.
+ * Class that may be used to configure various properties of cookies used for
+ * session tracking purposes.
  */
 public class SessionCookieConfigImpl implements SessionCookieConfig {
```
[the remark this answers is not in the document]
Comment (jGauravGupta) [reply appserver/web/web-core/src/main/java/org/apache/catalina/core/SessionCookieConfigImpl.java:62]: Added lazy initialization of map.

**Pull request merged**